### DISCUSSION TITLE
Overwrite plugin options and LSP server configurations

### DISCUSSION THREAD

**Discussion by appelgriebsch** [nvim-java/nvim-java#88] [Q&A] [answered]
Hey, 

I tried to setup an extension for LazyVim (https://github.com/appelgriebsch/Nv/blob/feature/nvim-java/lua/plugins/extras/lang/java.lua) and I want to understand:

- how to overwrite some of the options for the extension (like those https://github.com/nvim-java/nvim-java/blob/main/lua/java/config.lua)
- how to specify LSP server settings (like those https://github.com/appelgriebsch/Nv/blob/main/lua/plugins/lang.lua#L87)

Anyone an idea or link to some example?

**Comment by s1n7ax** [comment, marked as answer]
1.  you can override config in `config.lua` by calling `require('java').setup({java_test = { enabled = true } })`
2. You can change the server settings the same way you change/add the java runtimes https://github.com/nvim-java/nvim-java?tab=readme-ov-file#clamp-how-to-use-jdk-xx-version If you are using neoconf and a language server for json, then you will get autocompletion too for the configurations.

**Comment by appelgriebsch** [comment]
Hey, I tried something like this, but it doesn't seem to work as it still installs the `openjdk-17` from the mason registry:

```
      require("java").setup({
                jdk = {
                  auto_install = false,
                },
      })
```

**Reply by s1n7ax** [reply]
This works as expected. In lazy, when you pass a value to `opts = {}` then it automatically calls the setup function. That's probably what's happening. So either remove `opts={}` or remove the setup statement you have added and pass the configuration to `opts` property
